### PR TITLE
Data parser

### LINKED ISSUE #1: Parser

### PR DESCRIPTION
**Pull request by JohnLe01**
Closes #1 
- Create a parser that takes in the raw opal data and returns an object in the following format:

```
{
  diagnostic_entries = {
    1: "00,00, ...",
    ...
  },
  temperature_history = {
     1: {
      recorded: {
        date: "11 Oct 1989",
        time: "7:22 AM",
        deduced_seconds_elapsed:  ##
      },
      temperature: ##
    },
    ...
  }
 }
```

### CODE REVIEW

**Comment by JohnLe01** [comment]
Currently about 90% done. Still need to add a few more test cases and figure out how to do deduced_seconds_elapsed and refactor temperature output

**Comment by JohnLe01** [comment]
@gvela024 Update: 93% Done!! Managed to figure out regex for temperature. Still need to add 1-2 more test cases and figure out how to calculate deduced_seconds_elapsed

**Comment by JohnLe01** [comment]
And I just noticed that in First-Build-Mobile/First-Build/FSTOpal.m, the checkCompileOpalLogComplete function reads and appends FSTCharacteristicOpalLog1 to the string twice instead of reading and appending FSTCharacteristicOpalLog0 (Lines 812-816).. not sure if that's an accident or intended. Maybe that's why the first 224 digits in the data dump are all 0 lol

File: package.json [hunk 1]
```diff
@@ -10,5 +10,8 @@
   "license": "MIT",
   "devDependencies": {
     "jasmine": "^2.5.2"
+  },
+  "jshintConfig": {
+    "esversion": 6
```
Comment (gvela024) [inline package.json:15]: Is this a linter?

**Comment by gvela024** [comment]
Nice!!

Yeah, I think we should probably mark it as an issue in their code.

Sorry I've been MIA, worked on my part a little while this morning but then had to go run errands. Will try and put something up this afternoon.

File: spec/DataParser_spec.js [hunk 2]
```diff
@@ -1,8 +1,151 @@
 var DataParser = require('../src/DataParser');
 
 describe('Data Parser', function() {
-  it('should fail this test', function() {
-    DataParser();
-    expect(true).toBe(false);
+  it('should return empty object for empty data', function() {
+    var raw_data = new DataParser("");
```
Comment (gvela024) [inline spec/DataParser_spec.js]: I know I told you that we should do `snake_case`, but I think the general JS formatting is to use `camelCase` for variables and functions. My bad.

File: package.json (hunk 1 above)
Comment (JohnLe01) [reply package.json:15]: Yes, it's config stuff for the JS linter package i'm using with my atom 👍

File: spec/DataParser_spec.js (hunk 2 above)
Comment (JohnLe01) [reply spec/DataParser_spec.js]: Ah alrighty. either way is fine with me!

**Comment by JohnLe01** [comment]
It's fine! no rush or anything. Was just eager to work on this cause I wanted to get some more practice with JS and regex cause it's still all magic to me lol

File: package.json (hunk 1 above)
Comment (gvela024) [reply package.json:15]: Oh, ok nice, I think I need something like this too.

Comment (JohnLe01) [reply package.json:15]: first time using a linter.. it's really helpful

**Comment by JohnLe01** [comment]
Todo:
- Calculate seconds based on frequency of date and time in temperature data

**Comment by gvela024** [comment]
I think I'm giving up on the deduced seconds for now. Created #6 in case we find the time to do it.

**Pull request merged**